src/liballoc/collections/btree/node.rs

```diff
@@ -116,8 +116,11 @@ impl<K, V> LeafNode<K, V> {
         LeafNode {
             // As a general policy, we leave fields uninitialized if they can be, as this should
             // be both slightly faster and easier to track in Valgrind.
-            keys: MaybeUninit::uninitialized(),
-            vals: MaybeUninit::uninitialized(),
+            // Creating a `[MaybeUninit; N]` array by first creating a
+            // `MaybeUninit<[MaybeUninit; N]>`; the `into_inner` is safe because the inner
+            // array does not require initialization.
+            keys: MaybeUninit::uninitialized().into_inner(),
```

Comment:
is it possible to wrap it into a `MaybeUninit::uninitialized_array() -> [MaybeUninit; #]` static method that has the pre-copy as a doc-comment so as to remove the 7 times that exact comment is given? or is that (blocked by) what Centril meant by #49147 ?

Reply:
As I mentioned above, this can be made safe once #49147 is stable. Once that happens, you can just write `[MaybeUninit::uninitialized(), N]` directly.

Reply:
@fbstj You could make a `fn cap_uninit_array<T>() -> [MaybeUninit<T>; N]` function locally here for some specific `N`s (e.g. `CAPACITY`) using the unsafe method but you need const generics to do it for all `const N: usize`. So waiting for #49147 seems better.

Reply:
> why do I have a terrible feeling I'm going to be reading MaybeUninit::uninitialized().into_inner() everywhere rather than mem::uninitialized()...

In my defense, that's not the only change here. ;)

The main problem is that we have pretty bad support for working with arrays, and arrays of `MaybeUninit` are no exceptions. If we had functions generic over array lengths, this all would be much easier.

Reply:
Yeah-- it's just that working with uninitialized arrays is my only current use of `mem::uninitialized`, and it used to use `ManuallyDrop` etc. but that route is now broken. Seeing this working-around of `MaybeUninit`'s semantics makes me sad, but I understand it's a stepping-stone to the eventual goal.